plugins/queue/put.js

```diff
@@ -0,0 +1,502 @@
+'use strict';
+
+const logger = require('screwdriver-logger');
+const { merge, reach } = require('@hapi/hoek');
+const cron = require('./utils/cron');
+const Resque = require('node-resque');
+const helper = require('../helper');
+const { timeOutOfWindows } = require('./utils/freezeWindows');
+const DEFAULT_BUILD_TIMEOUT = 90;
+const RETRY_LIMIT = 3;
+const RETRY_DELAY = 5;
+
+/**
+ * Posts a new build event to the API
+ * @method postBuildEvent
+ * @param {Object} config           Configuration
+ * @param {Number} [config.eventId] Optional Parent event ID (optional)
+ * @param {Object} config.pipeline  Pipeline of the job
+ * @param {Object} config.job       Job object to create periodic builds for
+ * @param {String} config.apiUri    Base URL of the Screwdriver API
+ * @return {Promise}
+ */
+async function postBuildEvent(executor, { pipeline, job, apiUri, eventId, buildId, causeMessage }) {
+    const admin = await helper.getPipelineAdmin(
+        { redisBreaker: executor.redisBreaker, buildId },
+        apiUri,
+        pipeline.id);
+
+    logger.info(`POST event for pipeline ${pipeline.id}:${job.name} using user ${admin.username}`);
+
+    const buildEvent = {
+        pipelineId: pipeline.id,
+        startFrom: job.name,
+        creator: {
+            name: 'Screwdriver scheduler',
+            username: 'sd:scheduler'
+        },
+        causeMessage: causeMessage || 'Automatically started by scheduler'
+    };
+
+    if (eventId) {
+        buildEvent.parentEventId = eventId;
+    }
+    await helper.createBuildEvent(
```

Comment:
we should apply fix from https://github.com/screwdriver-cd/executor-queue/pull/82/files here

Reply:
cool!